src/Our.Umbraco.InnerContent/Web/UI/App_Plugins/InnerContent/js/innercontent.js

```diff
@@ -33,23 +37,26 @@ angular.module("umbraco").controller("Our.Umbraco.InnerContent.Controllers.DocTy
 
         innerContentService.getAllContentTypes().then(function (docTypes) {
             vm.docTypes = docTypes;
+            updateSelectedDocTypes();
         });
 
         if (!$scope.model.value) {
             $scope.model.value = [];
-            add();
         }
 
         function add() {
-            $scope.model.value.push({
+            var newItem = {
                 icContentTypeGuid: "",
                 nameTemplate: ""
-            });
+            };
+            $scope.model.value.push(newItem);
```

Comment:
If a user adds a new doctype, then presses Cancel, the un-configured row is still there. Maybe we should push to the array once the user has selected a doctype? (e.g. in the `submit` function)

Reply:
Yeah, I like that idea.